### PR TITLE
include: split uapi headers into more directories

### PR DESCRIPTION
**Pull request by jajanusz**
As discussed in: #1471 

Some of interface headers are not needed by kernel,
but are needed by user-space applications, so we can split them
into multiple directories that indicate their purpose.

Signed-off-by: Janusz Jankowski <janusz.jankowski@linux.intel.com>

### CODE REVIEW

**Comment by jajanusz** [comment]
@plbossart I made kernel folder to reflect what's there: https://github.com/thesofproject/linux/tree/topic/sof-dev/include/uapi/sound/sof

I guess that all headers (excluding ipc) that kernel needs.

File: CODEOWNERS [hunk 1]
```diff
@@ -11,7 +11,9 @@
 # include files
 src/include/sof/dmic.h			@singalsu
 src/include/host/*			@ranj063
-src/include/uapi/**			@thesofproject/steering-committee
+src/include/ipc/**			@thesofproject/steering-committee
+src/include/kernel/**			@thesofproject/steering-committee
+src/include/user/**			@thesofproject/steering-committee
```
Comment (plbossart) [inline CODEOWNERS]: Question: what does the "**" do here? Is this better or different than a single "*"?

Comment (jajanusz) [reply CODEOWNERS]: `*` matches only one level of files. If you want to match whole subdir tree you have to use `**`

**Comment by lgirdwood** [comment]
@zrombel will CI be available for this today or is it offline now until Monday ?

**Comment by tlauda** [comment]
@jajanusz There are some conflicts.

**Comment by jajanusz** [comment]
> @jajanusz There are some conflicts.

Fixed

**Pull request merged**